gemini/function-calling/forced_function_calling.ipynb

```diff
@@ -1,817 +1,830 @@
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "dmWOrTJ3gx13"
+   },
+   "source": [
+    "### Authenticate your notebook environment (Colab only)\n",
+    "\n",
+    "If you are running this notebook on Google Colab, run the cell below to authenticate your environment."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "id": "NyKGtVQjgx13"
+   },
+   "outputs": [],
+   "source": [
+    "import sys\n",
+    "\n",
+    "if \"google.colab\" in sys.modules:\n",
+    "    from google.colab import auth\n",
+    "\n",
+    "    auth.authenticate_user()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "DF4l8DTdWgPY"
+   },
+   "source": [
+    "### Set Google Cloud project information and initialize Vertex AI SDK\n",
+    "\n",
+    "To get started using Vertex AI, you must have an existing Google Cloud project and [enable the Vertex AI API](https://console.cloud.google.com/flows/enableapi?apiid=aiplatform.googleapis.com).\n",
+    "\n",
+    "Learn more about [setting up a project and a development environment](https://cloud.google.com/vertex-ai/docs/start/cloud-environment)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "id": "Nqwi-5ufWp_B"
+   },
+   "outputs": [],
+   "source": [
+    "PROJECT_ID = \"[your-project-id]\"  # @param {type:\"string\"}\n",
+    "LOCATION = \"us-central1\"  # @param {type:\"string\"}\n",
+    "\n",
+    "import vertexai\n",
+    "\n",
+    "vertexai.init(project=PROJECT_ID, location=LOCATION)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "23a24049e443"
+   },
+   "source": [
+    "## Import libraries"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "id": "cc6278ff6e55"
+   },
+   "outputs": [],
+   "source": [
+    "from IPython.display import Markdown, display\n",
+    "import arxiv\n",
+    "from vertexai.generative_models import (\n",
+    "    Content,\n",
+    "    FunctionDeclaration,\n",
+    "    GenerationConfig,\n",
+    "    GenerativeModel,\n",
+    "    Part,\n",
+    "    Tool,\n",
+    ")\n",
+    "from vertexai.preview.generative_models import ToolConfig"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "b1b86cbda668"
+   },
+   "source": [
+    "## Initialize model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "d091edc81048"
+   },
+   "source": [
+    "Initialize the Gemini model. Refer to the [Gemini Function Calling documentation](https://cloud.google.com/vertex-ai/generative-ai/docs/multimodal/function-calling) for more information on which models and model versions support forced function calling and tool configurations."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "id": "c8922eacdb2d"
+   },
+   "outputs": [],
+   "source": [
+    "model = GenerativeModel(\n",
+    "    \"gemini-1.5-pro-002\",\n",
```

Comment:
Do any of these explicitly require a numbered version?

We've been trending towards using the auto-updating version for notebooks `gemini-1.5-pro`

Reply:
Thanks for the reminder about this, I noticed that across other notebooks. It comes from habit of dealing with early versions and flakiness in Function Calling and Reasoning Engine. But things have stabilized enough for now that I'll flip all of these to `gemini-1.5-pro`.